pkg/controllers/raycluster_controller.go

```diff
@@ -544,6 +550,39 @@ func desiredHeadNetworkPolicy(cluster *rayv1.RayCluster, cfg *config.KubeRayConf
 		)
 }
 
+func (r *RayClusterReconciler) checkAndDeleteHeadPod(ctx context.Context, cluster *rayv1.RayCluster) error {
+	serviceAccount, err := r.kubeClient.CoreV1().ServiceAccounts(cluster.Namespace).Get(ctx, oauthServiceAccountNameFromCluster(cluster), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get OAuth ServiceAccount: %w", err)
+	}
+
+	headPod, err := getHeadPod(ctx, r, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to get head pod: %w", err)
+	}
+
+	if len(headPod.Spec.ImagePullSecrets) < 1 && len(serviceAccount.ImagePullSecrets) > 0 {
```

Comment:
It's probably more robust to check all the secrets from the ServiceAccount are present in the Pod.